### ISSUE TITLE
Accessing android glue (C functions) via a dependency.

### ISSUE THREAD

**Issue by tomc1998** [tomaka/android-rs-glue#120]
As far as I know, cargo apk links in the android glue C code into the project when being compiled, and can be accessed with the cargo_apk_injected_glue crate. However, if I wish to create a rust library to be linked into an APK at a later date, if I list the library as a dependency in the Cargo.toml file it won't have the cargo_apk_injected_glue crate available to it.

I have gotten around this currently by cloning the project and copying ffi.rs and lib.rs from cargo-apk/injected_glue, and creating a custom sub project. I can then access this by adding this subproject as a dependency.

Is there a way for this to be remedied, or am I doing stuff wrong?

**Comment by tomaka** [comment]
I'm not sure that's what you mean, but the [android_glue](https://crates.io/crates/android_glue) crate is maybe what you're looking for.
Its content is this: https://github.com/tomaka/android-rs-glue/blob/master/glue/src/lib.rs

**Comment by tomc1998** [comment]
Yeah I know, but I can't access the ffi function from this (i.e __android_write_log).

I can access the ffi functions from cargo_apk-injected_glue because that's linked into the project when building via `cargo apk`, but when I try and write a library intended for android `cargo apk` does not link the cargo_apk_injected_glue crate with the project dependencies, so I can't wrap the log function. I'm stuck with the log function from android_glue (`write_log()`) which doesn't let me specify a tag or priority.

Merry christmas dude!
